### PR TITLE
Refactor(input model): create the input

### PR DESCRIPTION
**Pull request by grysgreat**
rebuild class and function. This has no effect on the current code.
 Refactor #2549

### CODE REVIEW

**Comment by grysgreat** [comment]
Refact the input model. - Gu Hongyaoxing

File: source/module_io/parameter_pool.h
Comment (mohanchen) [inline source/module_io/parameter_pool.h:1]: Use English notes.

File: source/module_io/parameter_pool.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            ~param_value(){};
+
+        } value; // ����ֵ
+
+        InputParameter(){
+        }
+        /**
+         * @brief ���캯���������ò������ͺ�Ĭ�ϲ���ֵ
+         * 
+         * @param t ��������ö��ֵ
```
Comment (mohanchen) [inline source/module_io/parameter_pool.h]: Use English for all of the notes in ABACUS

**Review by mohanchen — CHANGES_REQUESTED** [review]

File: source/module_io/parameter_pool.cpp [hunk 2]
```diff
@@ -0,0 +1,1432 @@
...
+    else
+    {
+        std::cout << "Cannot open file !" << std::endl;
+    }
+}
+/**
+ * @brief This function is used to read the input parameter file and store it as a key-value pair
+ * @param fn Enter the path to the parameter file
+ */
+bool input_parameters_get(const std::string& fn)
```
Comment (hongriTianqi) [inline source/module_io/parameter_pool.cpp:71]: Please add unit tests for all the newly added functions.

File: source/module_io/parameter_pool.cpp [hunk 3]
```diff
@@ -0,0 +1,1432 @@
...
+ */
+extern map<std::string, InputParameter> input_parameters;
+extern map<std::string, string> default_parametes_type;
+
+/**
+ * @brief Converts the string sa to a minor character and stores it in the string sb
+ * @param sa Pointer to the string to be converted
+ * @param sb Store a pointer to the converted string, which needs to be pre-allocated with enough space
+ */
+void strtolower(char* sa, char* sb)
```
Comment (hongriTianqi) [inline source/module_io/parameter_pool.cpp:29]: Please write the new functions inside the namespace ModuleIO

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Review by hongriTianqi — CHANGES_REQUESTED** [review]
Please remove the empty git file

**Pull request closed without merging**